build-setup.sh

```diff
@@ -14,28 +14,63 @@ common_setup
 usage() {
     echo "Usage: ${0} [OPTIONS] [riscv-tools | esp-tools]"
     echo ""
-    echo "Helper script to initialize repository that wraps other scripts."
-    echo "Sets up conda environment, initializes submodules, and installs toolchain collateral."
-    echo ""
     echo "Installation Types"
     echo "  riscv-tools: if set, builds the riscv toolchain (this is also the default)"
     echo "  esp-tools: if set, builds esp-tools toolchain used for the hwacha vector accelerator"
     echo ""
+    echo "Helper script to fully initialize repository that wraps other scripts."
+    echo "By default it initializes/installs things in the following order:"
+    echo "   1. Conda environment"
+    echo "   2. Chipyard submodules"
+    echo "   3. Toolchain collateral (Spike, PK, tests, libgloss)"
+    echo "   4. Ctags"
+    echo "   5. Chipyard pre-compile sources"
+    echo "   6. FireSim"
+    echo "   7. FireSim pre-compile sources"
+    echo "   8. FireMarshal"
+    echo "   9. FireMarshal pre-compile br-base sources"
+    echo "  10. Runs repo. clean-up"
+    echo ""
+    echo "**See below for options to skip parts of the setup. However, they are not guaranteed to be tested/working.**"
+    echo ""
     echo "Options"
-    echo "  --help -h              : Display this message"
-    echo "  --unpinned-deps -ud    : Use unpinned conda environment"
-    echo "  --force -f             : Skip prompt checking for tagged release/conda"
-    echo "  --skip-validate        : DEPRECATED: Same functionality as --force"
-    echo "  --skip-conda           : Skip conda env creation"
-    echo "  --skip-toolchain-extra : Skip building extra RISC-V toolchain collateral (Spike, PK, tests, libgloos)"
+    echo "  --help -h                  : Display this message"
+
+    echo "  --force -f                 : Skip all prompts and checks"
+    echo "  --skip-validate            : DEPRECATED: Same functionality as --force"
+    echo "  --verbose                  : Verbose printout"
+
+    echo "  --use-unpinned-deps -ud    : Use unpinned conda environment"
+
+    echo "  --skip-conda               : Skip (1) conda env creation"
+
+    echo "  --skip-toolchain-extra     : Skip (3) building toolchain-specific extra RISC-V toolchain collateral"
+
+    echo "  --skip-cy-precompile       : Skip (5) Chipyard pre-compile SBT sources"
+
+    echo "  --skip-firesim             : Skip (6) (7) FireSim setup and pre-compile SBT sources"
+    echo "  --skip-firesim-precompile  : Skip (7) FireSim pre-compile SBT sources"
+
+    echo "  --skip-firemarshal         : Skip (8) (9) FireMarshal setup and pre-compile br-base sources"
+    echo "  --skip-firemarshal-br-base : Skip (9) FireMarshal pre-compile br-base sources"
+
+    echo "  --skip-repo-clean          : Skip (10) repository cleanup"
+
```

Comment:
This is really nice. Feature requests :pray: : 
 1. Allow skipping submodule-init and ctags (steps 2/4)
 2. Provide a generally flexible flag `-s n`